pages/index.tsx

```diff
@@ -1,281 +1,45 @@
-import { useRef, useState, useEffect } from 'react';
-import styles from '@/styles/Home.module.css';
-import { Message } from '@/types/chat';
-import Image from 'next/image';
-import ReactMarkdown from 'react-markdown';
-import LoadingDots from '@/components/ui/LoadingDots';
-import { Document } from 'langchain/document';
-import {
-  Accordion,
-  AccordionContent,
-  AccordionItem,
-  AccordionTrigger,
-} from '@/components/ui/accordion';
+import React, { useState } from 'react';
+import { Message, MessageState } from '@/types/chat';
 import Layout from '@/components/layout';
+import { MessageInput } from '@/components/MessageInput';
+import { MessageContainer } from '@/components/MessageContainer';
+
+const initMessages = (): Message[] => [
+  {
+    message: `Hi! I'm the (unofficial) Electoral Commission chatbot, I search guidance documents so you don’t have to`,
+    type: 'apiMessage',
+  },
+  {
+    message: `Remember the answers I provide are **not** legal advice. All data provided should be fact checked with The [Electoral Commission](https://www.electoralcommission.org.uk) tel: 0333 103 1928  
+    ![An anime meme where AI is asking if a butterfly is a pigeon](/ai-in-real-life.png)`,
+    type: 'apiMessage',
+  },
+  {
+    message: `Ask me a question! Here are some suggestions:  
+
+- How much money can I spend before I have to register?  
+- What are the spending limits for political parties?  
+- Do I need to register if I'm a charity distributing leaflets during an election?`,
+    type: 'apiMessage',
+  },
+];
 
 export default function Home() {
-  const [query, setQuery] = useState<string>('');
   const [loading, setLoading] = useState<boolean>(false);
-  const [error, setError] = useState<string | null>(null);
-  const [messageState, setMessageState] = useState<{
-    messages: Message[];
-    pending?: string;
-    history: [string, string][];
-    pendingSourceDocs?: Document[];
-  }>({
-    messages: [
-      {
-        message: `Hi!  
-        **I search** The Electoral Commission documents so you don’t have to  
-        Please remember the answers I provide are not legal advice, I am an experimental search tool. All data provided should be fact checked with The [Electoral Commission](https://www.electoralcommission.org.uk) tel: 0333 103 1928?
-        ![An anime meme where AI is asking if a butterfly is a pigeon](/ai-in-real-life.png)  
-        Ask me a question!  
-        Here are some suggestions:  
-        - How much money can I spend before I have to register?  
-        - What are the spending limits for political parties?  
-        - Do I need to register if I'm a charity distributing leaflets during an election?`,
-        type: 'apiMessage',
-      },
-    ],
+  const [messageState, setMessageState] = useState<MessageState>({
+    messages: initMessages(),
     history: [],
   });
 
-  const { messages, history } = messageState;
```

Comment:
Almost everything deleted from this file has just been moved into either `MessageInput` or `MessageContainer`.